src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/StorePal.cs

```diff
@@ -301,12 +301,14 @@ private static void LoadMachineStores()
             Volatile.Write(ref s_machineRootStore, rootStorePal);
         }
 
-        private static IEnumerable<T> Append<T>(IEnumerable<T> current, T addition)
+        private static IEnumerable<T> Prepend<T>(IEnumerable<T> current, T addition)
```

Comment:
One thing to consider is to use something like `trustedCertFiles = new [] { addition }.Union(trustedCertFiles)` instead.